src/v/storage/compaction_reducers.cc

```diff
@@ -292,7 +292,7 @@ ss::future<ss::stop_iteration> copy_data_segment_reducer::filter_and_append(
     const auto records_to_remove = record_count_before
                                    - to_copy->record_count();
     _stats.records_discarded += records_to_remove;
-    bool compactible_batch = is_compactible(_ntp, to_copy.value().header());
+    bool compactible_batch = is_compactible(to_copy.value().header());
```

Comment:
> However, `tx_fence` batches should be considered safely removable during
compaction, as the presence of the transactional bit in a Raft data batch
can accomplish the same "fencing" marker as the `tx_fence` batch itself.

I"m thinking we should should add some targeted tests for this scenario in ducktape ..  [<fence>..] [roll] [...data] [commit]/[abort]...  -> compact -> reboot -> recover learner  combinations..

I think this coverage maybe useful in the final validation of the feature too..